### ISSUE TITLE
Same Training Data on Resume

### ISSUE THREAD

**Issue by maaft** [fangchangma/sparse-to-dense.pytorch#19]
When creating the train-loader, every worker is initialized with its ID as a random seed.
```python
    if not args.evaluate:
        train_loader = torch.utils.data.DataLoader(
            train_dataset, batch_size=args.batch_size, shuffle=True,
            num_workers=args.workers, pin_memory=True, sampler=None,
            worker_init_fn=lambda work_id:np.random.seed(work_id))
```
When I decide to train the model for more than 15 epochs (i.e. with the --resume and --epochs arguments), the `work_id`s will be the same as in the first training, leading to essentially training the network with the same data *twice*.

I would suggest to modify above code e.g. to:
```python
worker_init_fn=lambda work_id:np.random.seed(work_id+args.epochs))
```
to generate **new** training data when *resuming* the training.

**Comment by fangchangma** [comment]
That's indeed a problem.  However, `work_id+args.epochs` is probably not the ideal solution because of possibly overlapping. For instance, work_id=20 at epochs=15 overlaps with work_id=15 at epochs=20 when resuming.

**Comment by maaft** [comment]
You are right.
What about
```python
worker_init_fn=lambda work_id:np.random.seed(work_id+args.workers*epoch))
```
where `epoch` is incremented on every epoch. This would of course require to rebuild the train_loader on each epoch unless there is a way to update the seeds during runtime:
I.e.:
```python
    def set_epoch(self, epoch):
        self.init_worker_fn = lambda work_id:np.random.seed(work_id + num_workers*epoch)
```
in `dataloader.py`

**Comment by fangchangma** [comment]
Preferably we don't have to rebuild the data loaders at every epoch. Additionally, this workaround might still suffer from seed overlapping if one resumes with a different number of workers.

**Comment by praful1993** [comment]
How to solve this error?
![Capture](https://user-images.githubusercontent.com/32381938/54083565-09f26700-434b-11e9-8131-91d3e868f4f4.JPG)
